pkg/dependencies/factory.go

```diff
@@ -423,7 +423,7 @@ func (f *Factory) WithHelm() *Factory {
 
 func (f *Factory) WithNetworking(clusterConfig *v1alpha1.Cluster) *Factory {
 	var networkingBuilder func() clustermanager.Networking
-	if clusterConfig.Spec.ClusterNetwork.CNI == v1alpha1.Kindnetd {
+	if clusterConfig.Spec.ClusterNetwork.CNIConfig != nil && clusterConfig.Spec.ClusterNetwork.CNIConfig.Kindnetd != nil {
```

Comment:
nit: if this has been validated, this shouldn't be necessary right? we can just check one